File: HW/HW10/HW10.java
public List<List<Integer>> levelOrder(TreeNode root) {
        List<List<Integer>> list = new LinkedList<>();
        Queue<TreeNode> queue = new LinkedList<>();
        queue.add(root);

        while(!queue.isEmpty()){
            List<Integer> subList = new LinkedList<>();
            int size = queue.size();

            for(int i = 0; i < size; i++){
                TreeNode node = queue.poll();
                if(node != null){
                    subList.add(node.val);
                    queue.add(node.left);
                    queue.add(node.right);
                }
            }
            
            if(!subList.isEmpty()){
                list.add(subList);
            }
        }

        return list;

}

File: HW/HW10/HW10.py
def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        list = []
        queue = collections.deque()
        queue.append(root)

        while queue:
            qLength = len(queue)
            subList = []

            for i in range(qLength):
                node = queue.popleft()
                if node:
                    subList.append(node.val)
                    queue.append(node.left)
                    queue.append(node.right)
            
            if subList:
                list.append(subList)
        
        return list
